File: Interview_Prep_App/native_java/retrofit/src/main/java/com/dreams/interviewprepapp/ui/score/ScoreActivity.java
package com.dreams.interviewprepapp.ui.score;


import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.dreams.interviewprepapp.R;
import com.dreams.interviewprepapp.ui.main.MainActivity;

/**
 * Created by Tc2r on 5/31/2017.
 * <p>
 * Description: Displays the score after a quiz is taken.
 */

public class ScoreActivity extends AppCompatActivity {

    // Extras Variables
    private int numCorrect;
    private int quizSize;
    private int scorePer;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Get Extras from intent
        Bundle extras = getIntent().getExtras();

        // Check if null
        if(extras != null) {

            // Assign Extras
            scorePer = extras.getInt("scorePercentage", 0);
            quizSize = extras.getInt("quizSize", 0);
            numCorrect = extras.getInt("numCorrect", 0);
        }
        setContentView(R.layout.activity_score);

        // Assign/initiate variables
        // UI Variables
        android.widget.TextView totalQuestionsTV = findViewById(com.dreams.interviewprepapp.R.id.numofquestions_tv);
        android.widget.TextView commentTV = findViewById(com.dreams.interviewprepapp.R.id.comment_tv);
        android.widget.TextView totalCorrectTV = findViewById(com.dreams.interviewprepapp.R.id.numcorrect_tv);
        android.widget.TextView finalGradeTV = findViewById(com.dreams.interviewprepapp.R.id.finalgrade_tv);


        //Set UI Objects Text values
        totalQuestionsTV.setText("There were " + quizSize + " Questions.");
        totalCorrectTV.setText("Correct Answers: " + numCorrect);
        finalGradeTV.setText(scorePer + "%");

        if(scorePer >= 80) {
            finalGradeTV.setTextColor(Color.GREEN);
            commentTV.setTextColor(Color.GREEN);
            commentTV.setText("OMEGA GOOD JOB!");
        } else if(scorePer > 50 && scorePer < 79) {
            finalGradeTV.setTextColor(Color.YELLOW);
            commentTV.setTextColor(Color.YELLOW);
            commentTV.setText("YOU'RE ALMOST THERE!");
        } else {
            finalGradeTV.setTextColor(Color.RED);
            commentTV.setTextColor(Color.RED);
            commentTV.setText("BETTER STUDY!");
        }
    }

    public void startOver(View view) {
        // Returns to MainActivity page.
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
    }

    public void closeApp(View view) {
        // Closes the App
        finishAffinity();
    }
}

File: Interview_Prep_App/native_kotlin/base/src/main/java/com/nudennie/interviewprepapp/ui/score/ScoreActivity.kt
package com.nudennie.interviewprepapp.ui.score

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.nudennie.interviewprepapp.R.id
import com.nudennie.interviewprepapp.R.layout
import com.nudennie.interviewprepapp.ui.main.MainActivity

/**
 * Created by Tc2r on 5/31/2017.
 *
 *
 * Description: Displays the score after a quiz is taken.
 */
class ScoreActivity : AppCompatActivity() {
    // UI Variables
    private var totalQuestionsTV: TextView? = null
    private var totalCorrectTV: TextView? = null
    private var finalGradeTV: TextView? = null
    private var commentTV: TextView? = null

    // Extras Variables
    private var numCorrect = 0
    private var quizSize = 0
    private var scorePer = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Get Extras from intent
        val extras = intent.extras

        // Check if null
        if (extras != null) {
            // Assign Extras
            scorePer = extras.getInt("scorePercentage", 0)
            quizSize = extras.getInt("quizSize", 0)
            numCorrect = extras.getInt("numCorrect", 0)
        }
        setContentView(layout.activity_score)

        // Assign/initiate variables
        totalQuestionsTV = findViewById<TextView>(id.numofquestions_tv)
        commentTV = findViewById<TextView>(id.comment_tv)
        totalCorrectTV = findViewById<TextView>(id.numcorrect_tv)
        finalGradeTV = findViewById<TextView>(id.finalgrade_tv)

        // Set UI Objects Text values
        totalQuestionsTV!!.text = "There were $quizSize Questions."
        totalCorrectTV!!.text = "Correct Answers: $numCorrect"
        finalGradeTV!!.text = "$scorePer%"
        if (scorePer >= 80) {
            finalGradeTV!!.setTextColor(Color.GREEN)
            commentTV!!.setTextColor(Color.GREEN)
            commentTV!!.text = "OMEGA GOOD JOB!"
        } else if (scorePer > 50 && scorePer < 79) {
            finalGradeTV!!.setTextColor(Color.YELLOW)
            commentTV!!.setTextColor(Color.YELLOW)
            commentTV!!.text = "YOU'RE ALMOST THERE!"
        } else {
            finalGradeTV!!.setTextColor(Color.RED)
            commentTV!!.setTextColor(Color.RED)
            commentTV!!.text = "BETTER STUDY!"
        }
    }

    fun startOver(view: View?) {
        // Returns to MainActivity page.
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
    }

    fun closeApp(view: View?) {
        // Closes the App
        finishAffinity()
    }
}
